api/src/main/java/org/openmrs/module/legacyui/GeneralUtils.java

```diff
@@ -68,4 +92,267 @@ public static Concept getConcept(String id) {
 		
 		return cpt;
 	}
+	
```

Comment:
@rubailly - to comment on this entire block of code in GeneralUtils:

* Anything that is an exact copy from a previous version of core that has been removed would be helpful to have a comment indicating where it came from.  (eg. url to file & line number in particular github branch)

* If methods were previously implemented as service methods with transactional annotations and privilege checking, we should retain this rather than just moving things into static methods that lose this.  For example "exitFromCare' and "discontinueAllDrugOrders" both make a series of calls to service methods to save data.  This will result in each of these occurring in separate, independent transactions rather than within one transaction, since the calling GeneralUtils method is not a transactional method.

* In order to sanity check these changes, I did so by comparing against the OpenMRS 1.12 branch as I assumed this is the last place where these methods were included in core.  But that doesn't match with the below.  I would think we should find the most recent version of these and use them, as they likely have bug fixes and other improvements.